libglusterfs/src/inode.c

```diff
@@ -933,6 +947,15 @@ __is_root_gfid(uuid_t gfid)
     return _gf_false;
 }
 
+static gf_boolean_t
+__is_root_with_ref(inode_t *inode)
+{
+    if (inode->ref && __is_root_gfid(inode->gfid))
```

Comment:
Why we do need to check the number of references ? the root inode will always have 1 reference at least (except on creation, but this function is not used whem created).

Reply:
The previous code used to check for it and hence I also added the same. Following our discussions, I know that root inode has ref < 1 only inside `inode_new`. We are safe to assume that ref will be > 1 everywhere else. I am removing this function then as it serves no purpose.